balancer/pickfirst/pickfirstleaf/pickfirstleaf.go

```diff
@@ -80,9 +102,12 @@ const (
 
 type pickfirstBuilder struct{}
 
-func (pickfirstBuilder) Build(cc balancer.ClientConn, _ balancer.BuildOptions) balancer.Balancer {
+func (pickfirstBuilder) Build(cc balancer.ClientConn, bo balancer.BuildOptions) balancer.Balancer {
 	b := &pickfirstBalancer{
-		cc:                    cc,
+		cc:              cc,
+		target:          bo.Target.String(),
+		metricsRecorder: bo.MetricsRecorder, // ClientConn will always create a Metrics Recorder so guaranteed to be non nil.
```

Comment:
This was Easwar's suggestion after our back and forth: https://github.com/grpc/grpc-go/pull/7839#discussion_r1849161049. What do you think about this. I'll drop "so guaranteed to be non nil" from this one because it's documented on the field.